webapp/src/js/services/enketo.js

```diff
@@ -525,6 +525,8 @@ angular.module('inboxServices').service('Enketo',
     };
 
     this.save = function(formInternalId, form, geolocation, docId) {
+      form.view.$.trigger('beforesave');
```

Comment:
I'm a little uncomfortable with this because `form.view.$` is not a public API so it could break without warning. I had a look in enketo-core to see if I could find another way and found their test app does this: https://github.com/enketo/enketo-core/blob/master/app.js#L56

Their implementation differs from this one in two ways:

1. They trigger this event after the validation. I think that makes sense so you don't end up calling it multiple times if validation fails the first time.
2. They trigger the event on the form element directly. We could also do this, eg: `$( selector + ' form.or' ).trigger( 'beforesave' );`. This means we're not using their internal APIs at all.

What do you think?

Reply:
That makes sense.